src/kvilib/net/KviSSL.cpp

```diff
@@ -828,8 +844,14 @@ void KviSSLCertificate::extractPubKeyInfo()
 	EVP_PKEY * p = X509_get_pubkey(m_pX509);
 	if(p)
 	{
+		int type;
+#if OPENSSL_VERSION_NUMBER >= 0x10100005L
+		type = EVP_PKEY_base_id(p);
+#else
+		type = EVP_PKEY_type(p->type);
+#endif
 		m_iPubKeyBits = EVP_PKEY_bits(p);
-		m_szPubKeyType = (p->type == NID_undef) ? __tr("Unknown") : OBJ_nid2ln(p->type);
+		m_szPubKeyType = (type == NID_undef) ? __tr("Unknown") : OBJ_nid2ln(type);
```

Comment:
Not sure about this one. There is no longer a way to get `p->type` - `EVP_PKEY_base_id(p)` replaces `EVP_PKEY_type(p->type)`, however I suspect that the original code was wrong and this was the intent.